pre_commit/commands/run.py

```diff
@@ -61,7 +61,11 @@ def get_filenames(args, include_expr, exclude_expr):
             lambda: get_changed_files(args.origin, args.source),
         )
     elif args.files:
-        getter = git.get_files_matching(lambda: args.files)
+        relative_files = [
+            os.path.relpath(os.path.abspath(filename), git.get_root())
+            for filename in args.files
+        ]
```

Comment:
I would just modify `args.files` in `pre_commit.main`, that'll make it less confusing for the rest of the program if it ever ends up using this argument in the future (and we reinvent this bug :D)